src/openrct2/scripting/ScEntity.hpp

```diff
@@ -323,18 +323,19 @@ namespace OpenRCT2::Scripting
             }
         }
 
-        ride_id_t ride_get() const
+        int32_t ride_get() const
         {
             auto vehicle = GetVehicle();
-            return vehicle != nullptr ? vehicle->ride : 0;
+            return static_cast<int32_t>(vehicle != nullptr ? vehicle->ride : RIDE_ID_NULL);
```

Comment:
should we increment plugin api for this change.

Reply:
I wasn't even sure if I should change that, but 0 is technically a valid ride in some cases.

Reply:
In JS you can return `undefined` or `null`, but you need to change the signature first. `std::optional` will work I think.

Reply:
I sort of think this should not be optional, you can set RIDE_ID_NULL but you would never be able to receive that value. I will do a version increment but return RIDE_ID_NULL instead, this is also a value that would be only returned on bad/unused vehicles I think.